src/Model/Payload.php

```diff
@@ -113,4 +110,39 @@ public function getCustomData(): array
     {
         return $this->customData;
     }
+
+    /**
+     * Guard the custom data upon instantiation.
+     *
+     * @param array $data
+     */
+    private function guardCustomData(array $data)
+    {
+        if (false === empty($data)) {
+            foreach ($data as $key => $value) {
+                $this->guardValue($value);
+                $this->guardKey($key);
+            }
+        }
+    }
+
+    private function guardValue($value)
+    {
+        if ($value && !is_array($value) && !is_scalar($value) && !$value instanceof \JsonSerializable) {
+            throw new \InvalidArgumentException(sprintf(
+                'The custom data value should be a scalar or \JsonSerializable instance, but "%s" given.',
+                is_object($value) ? get_class($value) : gettype($value)
+            ));
+        }
+    }
+
+    private function guardKey($key)
```

Comment:
Please rename method to `validateCustomDataKey` and add doc comment and return typehint (void) for this method.